pkg/image-manager/controller.go

```diff
@@ -140,7 +140,7 @@ func (h *imageEntityHandler) Delete(ctx context.Context, obj entitystore.Entity)
 		log.Error(err)
 	}
 
-	var deleted BaseImage
+	var deleted Image
```

Comment:
Very interesting error. I was thinking how a unit test to cover this bug would look like and why is it even needed. If the store could have a concrete type, this error wouldn't have happened. I feel like generating a concrete store interfaces for each entity using https://github.com/google/gvisor/blob/master/tools/go_generics/generics.go could help us increase confidence in the code.

Reply:
I kinda agree with having a concrete interface for storing entities to avoid such issues. Also, I realized this when I was trying to work with org entities which need to be stored and retrieved differently from other entities.